### PR TITLE
Fixed incorrect display of evasion chance in calc tab

### LINKED ISSUE #239: Evade chance in calcs not accounting for Raiders Avatar of the Chase
Chance to evade on the side correctly uses raiders "more chance to evade" but the calcs tab does not, as seen below
https://imgur.com/zcO29K5

### PR DESCRIPTION
**Pull request by meatballing**
There are 3 evasion chance calculations:
* evasion chance
* melee evasion chance
* projectile evasion chance

Evasion chance calculation does not change when there are modifiers to melee or projectile evasion chance. I have replaced it melee and projectile evasion chance and added remaining evasion relate mods to the mod list.

Closes #239 

image

### CODE REVIEW

File: Modules/CalcSections-3_0.lua [hunk 1]
```diff
@@ -972,9 +972,14 @@ return {
...
-		{ breakdown = "EvadeChance" },
-		{ modName = { "CannotEvade" } },
+	{ label = "Melee Evade Ch.", { format = "{0:output:MeleeEvadeChance}%", 
+		{ breakdown = "MeleeEvadeChance" },
+		{ modName = { "CannotEvade", "EvadeChance", "MeleeEvadeChance" } },
+		{ label = "Enemy modifiers", modName = { "Accuracy", "HitChance" }, enemy = true },
+	}, },
+	{ label = "Proj. Evade Ch.", { format = "{0:output:ProjectileEvadeChance}%", 
+		{ breakdown = "ProjectileEvadeChance" },
+		{ modName = { "CannotEvade", "EvadeChance", "ProjectileEvadeChance" } },
```
Comment (PJacek) [inline Modules/CalcSections-3_0.lua]:
```suggestion
		{ label = "Player modifiers", modName = { "CannotEvade", "EvadeChance", "ProjectileEvadeChance" } },
```

File: Modules/CalcSections-3_0.lua [hunk 2]
```diff
@@ -972,9 +972,14 @@ return {
 	{ label = "Total Increased", { format = "{0:mod:1}%", { modName = { "Evasion", "ArmourAndEvasion", "Defences" }, modType = "INC" }, }, },
 	{ label = "Total More", { format = "{0:mod:1}%", { modName = { "Evasion", "ArmourAndEvasion", "Defences" }, modType = "MORE" }, }, },
 	{ label = "Total", { format = "{0:output:Evasion}", { breakdown = "Evasion" }, }, },
-	{ label = "Evade Chance", { format = "{0:output:EvadeChance}%", 
-		{ breakdown = "EvadeChance" },
-		{ modName = { "CannotEvade" } },
+	{ label = "Melee Evade Ch.", { format = "{0:output:MeleeEvadeChance}%", 
+		{ breakdown = "MeleeEvadeChance" },
+		{ modName = { "CannotEvade", "EvadeChance", "MeleeEvadeChance" } },
```
Comment (PJacek) [inline Modules/CalcSections-3_0.lua]:
```suggestion
		{ label = "Player modifiers", modName = { "CannotEvade", "EvadeChance", "MeleeEvadeChance" } },
```

**Comment by NickRyder** [comment]
Unless I'm mistaken it seems like the two relevant stats are Melee Evasion Chance and Projectile Evasion Chance, so we should list both. (with evasion contributing to both)

**Comment by meatballing** [comment]
Updated the labels.

image
image

**Comment by Quotae** [comment]
In my opinion, it should only display two separate evasion chances if they're different. Otherwise it should display a single stat, which is how the sidebar currently works. You could maybe even add an Attack Avoid Chance stat for when melee and projectile avoid chances are the same. A program like PoB should display things in as few lines as possible and avoid redundant information so that everything's communicated as simply as it can be.

**Comment by meatballing** [comment]
Looking to get some advice on how to do this. In sidebar, there is a condFunc parameter that can be used to hide stats. Calc doesn't have this and I'm not too confident to implement a condFunc equivalent myself.

**Comment by meatballing** [comment]
Would it make sense to raise an feature enhancement to implement a condition function in calc and let this one change through for the time being?

**Pull request merged**